libnmstate/ifaces/base_iface.py

```diff
@@ -222,14 +222,15 @@ def pre_edit_validation_and_cleanup(self):
         check code.
         """
         if self.is_desired:
-            for family in (Interface.IPV4, Interface.IPV6):
-                self.ip_state(family).validate(
-                    IPState(family, self._origin_info.get(family, {}))
-                )
-            self._validate_port_ip()
-            ip_state = self.ip_state(family)
-            ip_state.remove_link_local_address()
-            self._info[family] = ip_state.to_dict()
+            if not self.is_absent:
+                for family in (Interface.IPV4, Interface.IPV6):
+                    self.ip_state(family).validate(
+                        IPState(family, self._origin_info.get(family, {}))
+                    )
+                self._validate_port_ip()
+                ip_state = self.ip_state(family)
+                ip_state.remove_link_local_address()
+                self._info[family] = ip_state.to_dict()
```

Comment:
Indeed, I think this applies for any kind of validation, right? When marking an interface as absent we should not validate anything.

Did you check if there are other validations being executed when the interface is being marked as absent?

Reply:
I have removed all the validations on absent interface. Please check the new patch.